manifests/init.pp

```diff
@@ -126,5 +132,19 @@
       group  => $puppet_group,
       mode   => '0750',
     }
+    if $fact_watcher_service {
+      if $manage_fact_watcher_dependencies {
+        package { ['ruby-inotify', 'parallel']:
+          ensure   => 'installed',
+          provider => 'puppet_gem',
+          before   => Systemd::Unit_file['fact_watcher.service'],
+        }
+      }
+      -> systemd::unit_file { 'fact_watcher.service':
```

Comment:
Can we use [systemd::manage_unit](https://github.com/voxpupuli/puppet-systemd/blob/master/manifests/manage_unit.pp)? I think this would work
```puppet
systemd::manage_unit { 'fact_watcher.service':
  enable => true,
  active => true,
  unit_entry => {
    'Description' => 'Publish facts to Foreman',
  },
  service_entry => {
    'Type' => 'simple',
    'Environment' => "PATH=/opt/puppetlabs/puppet/bin:${facts['path']}",
    'User' => $user,
    'ExecStart' => "${puppet_etcdir}/node.rb  --watch-facts --push-facts-parallel",
  },
  install_entry => {
    'WantedBy' => 'multi-user.target',
  },
}
```

Reply:
updated it to use systemd::manage_unit